network/p2p/compressed/compressedStream.go

```diff
@@ -74,6 +74,7 @@ func (c *compressedStream) Read(b []byte) (int, error) {
 func (c *compressedStream) Close() error {
 	c.writeLock.Lock()
 	defer c.writeLock.Unlock()
+	defer c.w.Close()
```

Comment:
Just to confirm, this means that we are swallowing any errors the this call might return. Are we sure that such errors are not of any interest (even for debugging purposes)? 
The `multierr` seemed a generally applicable solution, just wondering what your reasoning is for using a less-general approach.

Reply:
cc: @yhassanzadeh13

Reply:
@AlexHentschel as far as I investigated, closing the libp2p stream also closes the writer on it. However, here the `gzip` writer is a wrapper around the original writer of the libp2p stream. This poses a cyclic causality:
1. We close `gzip` writer (i.e., `c.w.Close()`) _prior_ to closing the stream (this was what the previous `multierr` was doing: `multierr.Combine(c.w.Close(), c.Stream.Close())`) . Then, with the new upgrade, once the `gzip` writer is closing, it also closes the underlying writer of the libp2p stream, which leaves the stream on a half-open status. Then when the original stream is closing (i.e., `c.Stream.Close()`) it fails to close as closing a half-open stream results in a `stream rest`, so it returns an error. 
2. We close `gzip` writer _after_ closing the original stream. Then `gzip` also tries closing the underlying writer of the libp2p stream. But since it is already closed it results in the closing an already closed stream, which again returns a `stream rest` error.

Hence, what matters is to first safely and gracefully close the libp2p stream (which also tells the remote node to free up its resource), and then invoke closing the `gzip` writer that was established on top of the libp2p stream's writer. Closing this `gzip` writer is with the goal of just cleaning up the resource locally making sure that there is no orphan resources lingering post stream closing. However, we are aware that closing that stream will cause a s `stream reset` error. 

To clarify this further, I added an error checking to check the returned error of closing `gzip` writer, and return it if it is not a `stream reset` error: https://github.com/onflow/flow-go/pull/3008